jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java

```diff
@@ -1,109 +1,109 @@
 package org.jbehavesupport.core.ssh;
 
+import static java.lang.Long.valueOf;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.nio.charset.Charset.defaultCharset;
+import static java.time.ZoneId.of;
+import static java.time.ZonedDateTime.now;
+import static java.time.format.DateTimeFormatter.ofPattern;
+import static net.schmizz.sshj.common.IOUtils.readFully;
 import static org.springframework.util.Assert.isTrue;
 import static org.springframework.util.Assert.notNull;
+import static org.springframework.util.StreamUtils.copyToString;
+import static org.springframework.util.StringUtils.isEmpty;
 
 import java.io.IOException;
-import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.List;
 
 import net.schmizz.sshj.SSHClient;
-import net.schmizz.sshj.common.IOUtils;
 import net.schmizz.sshj.connection.channel.direct.Session;
 import net.schmizz.sshj.transport.verification.PromiscuousVerifier;
 import net.schmizz.sshj.userauth.keyprovider.KeyProvider;
-import org.springframework.util.StringUtils;
+import org.springframework.core.io.ClassPathResource;
 
 public class SshTemplate {
-    private String timestampFormat;
-    private SshSetting sshSetting;
-
-    private RollingLogResolver rollingLogResolver;
+    private final SshSetting sshSetting;
+    private final String timestampFormat;
+    private final String timestampPosixRegex;
+    private final RollingLogResolver rollingLogResolver;
     private SSHClient sshClient;
 
     private interface Commands {
         String TIMEZONE = "timedatectl | grep zone | grep -Po '(.*zone: )\\K(\\S+)'";
         String EPOCH_TIME = "date +%s";
-        String FIRST_LAST_LINES_BETWEEN_TIMESTAMPS = "awk '{if ($1\" \"$2 >= \"%s\")  p=1; if ($1\" \"$2 >= \"%s\")  p=0;} p { print NR }' %s | sed -n '1p;$p'";
-        String PRINT_BETWEEN_LINES = "awk 'NR >= %d && NR <= %d' %s";
     }
 
     private interface Constants {
         String CMD_DELIMITER = "; ";
     }
 
-    public SshTemplate(SshSetting sshSetting, String timestampFormat, RollingLogResolver rollingLogResolver) {
+    public SshTemplate(SshSetting sshSetting,
+                       String timestampFormat, String timestampPosixRegex,
+                       RollingLogResolver rollingLogResolver) {
+        isTrue(!isEmpty(sshSetting.getLogPath()), "log path must not be null or empty");
```

Comment:
I think this is backward incompatible change, I'd keep the original version with 3 parameters and add the new one

Reply:
I pushed new solution which doesn't rely on timestampRegex, hence is backward compatible.